gemrb/core/Scriptable/Actor.cpp

```diff
@@ -514,19 +514,19 @@ void Actor::SetAnimationID(unsigned int AnimID)
 	// the rest had it hardcoded
 	if (!core->HasFeature(GF_RESDATA_INI)) {
 		// handle default speed and per-animation overrides
-		int row = -1;
+		TableMgr::index_t row = TableMgr::npos;
 		if (extspeed) {
 			char animHex[10];
 			snprintf(animHex, 10, "0x%04X", AnimID);
-			row = extspeed->FindTableValue((unsigned int) 0, animHex);
-			if (row != -1) {
-				int rate = atoi(extspeed->QueryField(row, 1));
+			row = extspeed->FindTableValue(0ul, animHex);
```

Comment:
Bill Gates prefers something like `static_cast<index_t>(0)` when expected to work on x86 and x64.